### PR TITLE
Restore state and null check to prevent crash returning to post preview

### LINKED ISSUE #5702: Crash on 7.0: NPE in PostPreviewActivity.onResume
http://crashes.to/s/4c6a3e19815

```
Fatal Exception: java.lang.RuntimeException: Unable to resume activity {org.wordpress.android/org.wordpress.android.ui.posts.PostPreviewActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.wordpress.android.fluxc.model.PostModel.getId()' on a null object reference
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3527)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3558)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1526)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:6145)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int org.wordpress.android.fluxc.model.PostModel.getId()' on a null object reference
       at org.wordpress.android.ui.posts.PostPreviewActivity.onResume(PostPreviewActivity.java:100)
       at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1255)
       at android.app.Activity.performResume(Activity.java:6495)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3516)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3558)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1526)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:6145)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

### PR DESCRIPTION
**Pull request by tonyr59h**
Fixes #5702 

`onRestoreInstanceState` wasn't implemented so the post and site were never restored. I've added that overload and included a null check at the crashing line just in case.

To test:
* Make a local draft
* Preview it
* Rotate, background, etc... and restore
* Verify no crashes

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java [hunk 1]
```diff
@@ -151,6 +156,17 @@ private void refreshPreview() {
     }
 
     @Override
+    protected void onRestoreInstanceState(Bundle savedState) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: The state is already restored in `onCreate()` so having `onRestoreInstanceState()` is redundant.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java [hunk 2]
```diff
@@ -97,6 +98,10 @@ protected void onResume() {
         EventBus.getDefault().register(this);
         mDispatcher.register(this);
 
+        if (mPost == null) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: Shouldn't this null check come after the `getPostByLocalPostId()` call below?

Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: :rofl: oops!

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: Just noticed this uses `mPost.getId()` to get the post again. Do we know `mPost` won't be null here?

**Comment by nbradbury** [comment]
This latest change fixes that specific crash, but I think the real problem (or, at least, a related problem) is we're not checking if `mPost` is null in `onCreate()`.

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**